spec/features/support/providers/creating_a_provider_spec.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+feature "Creating a provider" do
+  let(:user) { create(:user, :admin) }
+
+  before do
+    given_i_am_authenticated(user: user)
+    when_i_visit_the_new_provider_page
+  end
+
+  scenario "adds a new provider" do
+    when_i_fill_in_a_valid_provider_details
+    and_i_fill_in_a_valid_organisation_details
+    and_i_fill_in_a_valid_site_details
+    and_i_click_the_submit_button
+    then_i_am_redirected_back_to_the_provider_show_page
+    and_the_provider_is_created
+  end
+
+  context "invalid details" do
```

Comment:
I think the consensus was to avoid using `context` wraps and potentially too many nested ones in feature specs to keep things simple